mongo/integration/sessions_mongocryptd_prose_test.go

```diff
@@ -0,0 +1,169 @@
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/event"
+	"go.mongodb.org/mongo-driver/internal/assert"
+	"go.mongodb.org/mongo-driver/internal/require"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+type mongocryptdProcess struct {
+	cmd *exec.Cmd
+}
+
+// start will start a mongocryptd server in the background on the OS.
+func (p *mongocryptdProcess) start(port int) error {
+	args := []string{
+		"mongocryptd",
+		"--port", strconv.Itoa(port),
+	}
+
+	p.cmd = exec.Command(args[0], args[1:]...) //nolint:gosec
+	p.cmd.Stderr = p.cmd.Stdout
+
+	return p.cmd.Start()
+}
+
+// close will kill the underlying process on the command.
+func (p *mongocryptdProcess) close() error {
+	if p.cmd.Process == nil {
+		return nil
+	}
+
+	if err := p.cmd.Process.Kill(); err != nil {
+		return fmt.Errorf("failed to terminate mongocryptd process: %w", err)
+	}
+
+	// Release instead of wait to avoid blocking in CI.
+	err := p.cmd.Process.Release()
+	if err != nil {
+		return fmt.Errorf("failed to release mongocryptd: %w", err)
+	}
+
+	return nil
+}
+
+func TestSessionsMongocryptdProse(t *testing.T) {
+	const mongocryptdPort = 27022
+
+	// Monitor the lsid value on commands. If an operation run in any
+	// subtests contains an lsid, then the Go Driver wire message
+	// construction has incorrectly interpreted that
+	// LogicalSessionTimeoutMinutes was returned by the server on handshake.
+	cmdMonitor := &event.CommandMonitor{
+		Started: func(_ context.Context, evt *event.CommandStartedEvent) {
+			_, err := evt.Command.LookupErr("lsid")
+			assert.ErrorIs(t, err, bsoncore.ErrElementNotFound)
+		},
+	}
+
+	uri := &url.URL{
+		Scheme: "mongodb",
+		Host:   net.JoinHostPort("localhost", strconv.Itoa(mongocryptdPort)),
+	}
+
+	mtOpts := mtest.NewOptions().
+		MinServerVersion("4.2").
+		Topologies(mtest.ReplicaSet, mtest.Sharded).
+		CreateCollection(false).
+		CreateClient(false)
+
+	// Create a new instance of mtest (MongoDB testing framework) for this
+	// test and configure it to control server versions.
+	mt := mtest.New(t, mtOpts)
+	mt.Cleanup(mt.Close)
+
+	proc := mongocryptdProcess{}
+
+	// Start a mongocryptd server.
+	err := proc.start(mongocryptdPort)
+	require.NoError(t, err, "failed to create a mongocryptd process: %v", err)
+
+	t.Cleanup(func() {
+		err := proc.close()
+		require.NoError(t, err, "failed to close mongocryptd: %v", err)
+	})
+
+	clientOpts := options.
+		Client().
+		ApplyURI(uri.String()).
+		SetMonitor(cmdMonitor)
+
+	ctx := context.Background()
+
+	client, err := mongo.Connect(ctx, clientOpts)
```

Comment:
Optional: Is there a reason we can't use the client/collection created by `mt.RunOpts`? If so, consider adding that reason as a comment here that describes why we create a client not managed by `mtest`.